vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -321,15 +339,91 @@ and
 
     let optionsAssociation = ConditionalWeakTable<IWorkspaceProjectContext, string[]>()
 
+    member private this.ProvideProjectSiteProvider(workspace:Workspace, project:Project) =
+        let visualStudioWorkspace = workspace :?> VisualStudioWorkspace
+        let hier = visualStudioWorkspace.GetHierarchy(project.Id)
+
+        {new IProvideProjectSite with
+            member iProvideProjectSite.GetProjectSite() =
+                let sources () = 
```

Comment:
I think I mentioned this in a previous review - but it is clearer without these micro-functions, e.g. 

```fsharp
        {new IProvideProjectSite with
            member iProvideProjectSite.GetProjectSite() =
                let targetFrameworkMoniker = ""
                let creationTime = System.DateTime.Now
                let mutable errorReporter = 
                    let reporter = ProjectExternalErrorReporter(project.Id, "FS", this.SystemServiceProvider)
                    Some(reporter:> Microsoft.VisualStudio.Shell.Interop.IVsLanguageServiceBuildErrorReporter2)

                {new Microsoft.VisualStudio.FSharp.LanguageService.IProjectSite with
                    member __.SourceFilesOnDisk() =  projectInfoManager.GetProjectInfo(project.FilePath) |> p13
                    member __.DescriptionOfProject() = project.Name
                    member __.CompilerFlags() = 
                        let _,references,options = projectInfoManager.GetProjectInfo(project.FilePath)
                        Array.concat [options; references |> Array.map(fun r -> "-r:" + r)]
                    member __.ProjectFileName() = project.FilePath
                    member __.AdviseProjectSiteChanges(_,_) = ()
                    member __.AdviseProjectSiteCleaned(_,_) = ()
                    member __.AdviseProjectSiteClosed(_,_) = ()
                    member __.IsIncompleteTypeCheckEnvironment = false
                    member __.TargetFrameworkMoniker = targetFrameworkMoniker
                    member __.ProjectGuid = project.Id.Id.ToString()
                    member __.LoadTime = creationTime
                    member __.ProjectProvider = Some iProvideProjectSite
                    member __.AssemblyReferences() = projectInfoManager.GetProjectInfo(project.FilePath) |> p33
                    member __.BuildErrorReporter with get () = errorReporter and 
                                                      set (v) = errorReporter <- v
               }
```